js/models/SMTPSettings.js

```diff
@@ -21,7 +21,20 @@ export default class extends BaseModel {
       errObj[fieldName].push(error);
     };
 
-    if (attrs.notifications) {
+    // get all keys that have String values
+    const stringValuedKeys = Object.keys(attrs).filter(k => typeof attrs[k] === 'string');
+
+    // trim all string values of whitespace
+    stringValuedKeys.forEach(k => { attrs[k] = attrs[k].trim(); });
+
+    // check if they are all empty
+    const everyInputEmpty = stringValuedKeys.every(k => attrs[k].length === 0);
+
+    if (everyInputEmpty) {
```

Comment:
Rob's right, we should do one of the following here:

1. only validate the fields that have a value when notifications are turned off, so blank fields are not marked invalid but fields with data that is incorrect are.

2. go back to only validating when notifications are turned on. That would still allow saving of invalid data when that data isn't used, but that would be better than making the user think they need to fill in all the blank fields when notifications are off.